src/app.js

```diff
@@ -136,16 +136,18 @@ const App = Marionette.Application.extend({
 
     this.setRegion(opts.region);
 
-    this.triggerMethod('before:start', options);
-
     opts.state = this.getInitState(opts.state);
 
-    this._isRunning = true;
-
     this.initState(opts);
 
+    this.triggerMethod('before:start', options);
+
+    this._isRunning = true;
+
     this.triggerStart(opts);
 
+    this.delegateStateEvents();
```

Comment:
this isn't going to work.  [`triggerStart`](https://github.com/RoundingWellOS/marionette.toolkit/blob/master/docs/async-app-start.md) was added so that `onStart` could be called async, but this will be called no matter if `triggerStart` is made async or not.

However now I'm second guessing setting the state events after `onStart`.  It does seem reasonable to do so in `initialize` for Component..  it does still seem very useful to make the state model available to `onBeforeStart`, but now I'm think that making the `stateEvents` wait until after `_isRunning === true` is enough, but getting it to happen after all listeners of the `start` event seems hard and I'm not entirely sure it's a good idea anyhow.  Thoughts?